src/foremast/s3/s3deploy.py

```diff
@@ -79,25 +80,32 @@ def _path_formatter(self, suffix):
             str: formatted path
 
         """
-        path_format = "{}/{}/{}"
+        if suffix.lower() == "flat":
+            path_format = "{}/{}"
+            path = path_format.format(self.bucket, self.s3path)
+        else:
+            path_format = "{}/{}/{}"
+            path = path_format.format(self.bucket, self.s3path, suffix)
         s3_format = "s3://{}"
-        path = path_format.format(self.bucket, self.s3path, suffix)
         formatted_path = path.replace('//', '/')  # removes configuration errors
         full_path = s3_format.format(formatted_path)
         return full_path
 
     def upload_artifacts(self):
-        """Upload artifacts to S3 and copy to LATEST depending on strategy."""
+        """Upload artifacts to S3 and copy to correct path depending on strategy."""
         deploy_strategy = self.properties["deploy_strategy"]
-        self._upload_artifacts_to_version()
-        if deploy_strategy == "highlander":
-            self._sync_to_uri(self.s3_latest_uri)
-        elif deploy_strategy == "canary":
-            self._sync_to_uri(self.s3_canary_uri)
-        elif deploy_strategy == "alpha":
-            self._sync_to_uri(self.s3_alpha_uri)
+        if deploy_strategy == "flat":
```

Comment:
Why is this whole section not like:
```
if deploy_strategy == "x":
    ....
elif deploy_strategy == "y":
   ....
elif deploy_strategy == "z":
   ...
```

Reply:
because of the `  self._upload_artifacts_to_path(flat=False)` part should run before all the other non-flat deployments.

Reply:
Ahh. So no matter what `self._upload_artifacts_to_path(..)` needs to run. If that's so, then i think this would work:

```
flat = False
if deploy_strategy = 'flat':
   flat = True

# this runs against all strategies
self._upload_artifacts_to_path(flat=flat)

# run specific sync_uri function
if deploy_strategy == "x":
     self._sync_to_uri(self.s3_latest_uri)
elif deploy_strategy == "y":
    self._sync_to_uri(self.s3_canary)
elif deploy_strategy == "z":
   ....

```

Reply:
ahh yeah, I like that.